src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -185,6 +196,58 @@ protected int doWork() {
                     " already exists.  Delete this file and try again, or specify a different output file.");
         }
 
+        // map for aliases mapping a contig to its' aliases
+        final Map<String, Set<String>> contig2aliases = new HashMap<>();
+        
+        // fill the alias map
+        if(this.ALT_NAMES != null) {
+            try {
+                // regex defined in the sam spec
+                final Pattern altNameRegex = Pattern.compile("[0-9A-Za-z][0-9A-Za-z\\*\\+@\\|\\-]*");
+                for(final String line :IOUtil.slurpLines(this.ALT_NAMES)) {
+                    if(StringUtil.isBlank(line)) continue;
+                    final int tab = line.indexOf('\t');
+                    if(tab == -1 ) throw new IOException("tabulation missing in " + line);
+                    final String contigName = line.substring(0,tab);
+                    final String altName = line.substring(tab + 1);
+                    //check for empty values
+                    if(StringUtil.isBlank(contigName)) {
+                        throw new IOException("empty contig in  " + line);
+                    }
+                    if(StringUtil.isBlank(altName)) {
+                        throw new IOException("empty alternative name in  " + line);
+                    }
+                    if(altName.equals(contigName)) continue;
+                    if(!altNameRegex.matcher(altName).matches()) {
+                        throw new IOException("alternative name in  " + line +
+                                " doesn't match the regular expression : " +
+                                altNameRegex.pattern());
+                    }
+                    //check alias not previously defined as contig
+                    if(contig2aliases.containsKey(altName)) {
+                        throw new IOException("alternate name  " + altName +
+                                "previously defined as a contig in " + line);
+                    }
+                    //check contig not previously defined as alias
+                    if(contig2aliases.keySet().stream().
+                            filter(K->!K.equals(contigName)). // not an error is defined twice for same contig
+                            flatMap(K->contig2aliases.get(K).stream()).
```

Comment:
If you need to iterate over both keys and values, it's better to stream over `entrySet()` instead of looping over keys and doing a `get()` for each key.